webapp/graphite/tags/localdatabase.py

```diff
@@ -0,0 +1,239 @@
+import re
+
+from graphite.tags.utils import BaseTagDB, TaggedSeries
+from django.db import connection
+
+class LocalDatabaseTagDB(BaseTagDB):
+  def find_series_query(self, tags):
+    sql = 'SELECT s.path'
```

Comment:
Performance, this is avoiding all the python overhead that the ORM brings.

Reply:
Is performance really important when using the local database ? And the ORM isn't that bad.

Reply:
I'm not sure that it's even possible to implement the logic in `find_series_query` using the ORM, because of the complex joins it uses, but I'm not a django ORM expert.  The same is true for the multi-value insert ignore statements used in `tag_series`, which afaik in the ORM would have to be implemented with separate queries for each tag and each value.   If you know the ORM I'd be keen to see how it could be implemented, but I'm much more confident in saying that the raw SQL will work as correctly and efficiently as I can make it.

Reply:
That's reasonable for the first version, I could later try to port it to the ORM to see how it looks